src/Services/InsuranceCompanyService.php

```diff
@@ -61,21 +75,64 @@ public function getAll()
         $sql .= " FROM insurance_companies i";
         $sql .= " JOIN addresses a ON i.id = a.foreign_id";
 
-        $statementResults = sqlStatement($sql);
+        if (!empty($search)) {
+            $sql .= ' AND ';
+            $whereClauses = array();
+            foreach ($search as $fieldName => $fieldValue) {
+                array_push($whereClauses, $fieldName . ' = ?');
+                array_push($sqlBindArray, $fieldValue);
+            }
+            $sqlCondition = ($isAndCondition == true) ? 'AND' : 'OR';
+            $sql .= implode(' ' . $sqlCondition . ' ', $whereClauses);
+        }
+
+        $statementResults = sqlStatement($sql, $sqlBindArray);
 
-        $results = array();
+        $processingResult = new ProcessingResult();
         while ($row = sqlFetchArray($statementResults)) {
-            array_push($results, $row);
+            $row['uuid'] = UuidRegistry::uuidToString($row['uuid']);
+            $processingResult->addData($row);
         }
-
-        return $results;
+        return $processingResult;
     }
 
-    public function getOne($id)
+    public function getOne($uuid)
```

Comment:
migrate the old getOne function to getOneById (and change where it is called in the codebase; it is used 1 time in the codebase in receipts_by_method_report.php).

Reply:
Added a new function getOneById() , changed the same in codebase in receipts_by_method_report.php and InsuranceCompanyRestController.php

Reply:
It's also used in
interface/billing/sl_eob_process.php
I think everywhere else is covered but i'd search code base to double check me...